gcloud-java-dns/src/test/java/com/google/gcloud/dns/ManagedZoneInfoTest.java

```diff
@@ -0,0 +1,199 @@
+  private static final Long CREATION_TIME_MILLIS = 1123468321321L;
+  private static final String DNS_NAME = "example.com.";
+  private static final String DESCRIPTION = "description for the zone";
+  private static final String NAME_SERVER_SET = "some set";
+  private static final String NS1 = "name server 1";
+  private static final String NS2 = "name server 2";
+  private static final String NS3 = "name server 3";
+  private List<String> nameServers = new LinkedList<>();
+  private ManagedZoneInfo info;
+
+  @Before
+  public void setUp() {
+    nameServers.add(NS1);
+    nameServers.add(NS2);
+    nameServers.add(NS3);
+    info = ManagedZoneInfo.builder(NAME, ID)
+        .creationTimeMillis(CREATION_TIME_MILLIS)
+        .dnsName(DNS_NAME)
+        .description(DESCRIPTION)
+        .nameServerSet(NAME_SERVER_SET)
+        .nameServers(nameServers)
+        .build();
+  }
+
+  @Test
+  public void testDefaultBuilders() {
+    ManagedZoneInfo withName = ManagedZoneInfo.builder(NAME).build();
+    assertTrue(withName.nameServers().isEmpty());
+    assertEquals(NAME, withName.name());
+    assertNull(withName.id());
+    assertNull(withName.creationTimeMillis());
+    assertNull(withName.nameServerSet());
+    assertNull(withName.description());
+    assertNull(withName.dnsName());
+    ManagedZoneInfo withId = ManagedZoneInfo.builder(ID).build();
+    assertTrue(withId.nameServers().isEmpty());
+    assertEquals(ID, withId.id());
+    assertNull(withId.name());
+    assertNull(withId.creationTimeMillis());
+    assertNull(withId.nameServerSet());
+    assertNull(withId.description());
+    assertNull(withId.dnsName());
+    ManagedZoneInfo withBoth = ManagedZoneInfo.builder(NAME, ID).build();
+    assertTrue(withBoth.nameServers().isEmpty());
+    assertEquals(ID, withBoth.id());
+    assertEquals(NAME, withBoth.name());
+    assertNull(withBoth.creationTimeMillis());
+    assertNull(withBoth.nameServerSet());
+    assertNull(withBoth.description());
+    assertNull(withBoth.dnsName());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(3, info.nameServers().size());
+    assertEquals(NS1, info.nameServers().get(0));
+    assertEquals(NS2, info.nameServers().get(1));
+    assertEquals(NS3, info.nameServers().get(2));
+    assertEquals(NAME, info.name());
+    assertEquals(ID, info.id());
+    assertEquals(CREATION_TIME_MILLIS, info.creationTimeMillis());
+    assertEquals(NAME_SERVER_SET, info.nameServerSet());
+    assertEquals(DESCRIPTION, info.description());
+    assertEquals(DNS_NAME, info.dnsName());
+  }
+
+  @Test
+  public void testValidCreationTime() {
+    ManagedZoneInfo.builder(NAME).creationTimeMillis(-1);
+    ManagedZoneInfo.builder(NAME).creationTimeMillis(0);
+    ManagedZoneInfo.builder(NAME).creationTimeMillis(Long.MAX_VALUE);
+  }
+
+  @Test
+  public void testEqualsAndNotEquals() {
+    ManagedZoneInfo clone = info.toBuilder().build();
+    assertEquals(clone, info);
```

Comment:
Nit, here and below (I also saw that you did the same in DnsRecordTest):
assert's first argument should be the *expected* value.